lib/jerboa/client.ex

```diff
@@ -109,15 +109,24 @@ defmodule Jerboa.Client do
   """
   @spec allocate(t) :: {:ok, address} | {:error, error}
   def allocate(client) do
-    GenServer.call(client, :allocate)
+    call = fn -> GenServer.call(client, :allocate) end
+    case call.() do
+      {:error, :stale_nonce} -> call.()
+      {:error, :unauthorized} -> call.()
+      result -> result
+    end
   end
 
   @doc """
   Tries to refresh the allocation on the server
   """
   @spec refresh(t) :: :ok | {:error, error}
   def refresh(client) do
-    GenServer.call(client, :refresh)
+    call = fn -> GenServer.call(client, :refresh) end
```

Comment:
How about `maybe_retry(request/1, [:refresh])`? This block is used several times, so splitting it into two functions feels like a good idea. 

```elixir
defp request(req), do: GenServer.call(client, req)
defp maybe_retry(fun, req) do
  case fun(req) do
    {:error, :stale_nonce} -> fun(req)
    {:error, :unauthorized} -> fun(req)
    result -> result
  end
end